plugins/autocomplete/plugin.js

```diff
@@ -0,0 +1,1422 @@
+		 */
+		onData: function( evt ) {
+			if ( this.model.hasData() ) {
+				this.view.updateItems( evt.data );
+				this.open();
+			} else {
+				this.close();
+			}
+		},
+
+		/**
+		 * The function registered by the {@link #attach} method as the
+		 * {@link CKEDITOR.plugins.autocomplete.view#click-item} event listener.
+		 *
+		 * @param {CKEDITOR.eventInfo} evt
+		 */
+		onItemClick: function( evt ) {
+			this.commit( evt.data );
+		},
+
+		/**
+		 * The function registered by the {@link #attach} method as the `keydown` event listener.
+		 *
+		 * @param {CKEDITOR.dom.event} evt
+		 */
+		onKeyDown: function( evt ) {
+			if ( !this.model.isActive ) {
+				return;
+			}
+
+			var keyCode = evt.data.getKey(),
+				handled = false;
+
+			// Esc key.
+			if ( keyCode == 27 ) {
+				this.close();
+				this.textWatcher.unmatch();
+				handled = true;
+			// Down Arrow.
+			} else if ( keyCode == 40 ) {
+				this.model.selectNext();
+				handled = true;
+			// Up Arrow.
+			} else if ( keyCode == 38 ) {
+				this.model.selectPrevious();
+				handled = true;
+			// Completion keys.
+			} else if ( CKEDITOR.tools.indexOf( this.commitKeystrokes, keyCode ) != -1 ) {
+				this.commit();
+				this.textWatcher.unmatch();
+				handled = true;
+			}
+
+			if ( handled ) {
+				evt.cancel();
+				evt.data.preventDefault();
+				this.textWatcher.consumeNext();
+			}
+		},
+
+		/**
+		 * The function registered by the {@link #attach} method as the
+		 * {@link CKEDITOR.plugins.autocomplete.view#change-selectedItemId view#change-selectedItemId}
+		 * and {@link CKEDITOR.plugins.autocomplete.model#change-selectedItemId model#change-selectedItemId} event listeners.
+		 *
+		 * @param {CKEDITOR.eventInfo} evt
+		 */
+		onSelectedItemId: function( evt ) {
+			this.model.setItem( evt.data );
+			this.view.selectItem( evt.data );
+		},
+
+		/**
+		 * The function registered by the {@link #attach} method as the
+		 * {@link CKEDITOR.plugins.textWatcher#matched} event listener.
+		 *
+		 * @param {CKEDITOR.eventInfo} evt
+		 */
+		onTextMatched: function( evt ) {
+			this.model.setActive( false );
+			this.model.setQuery( evt.data.text, evt.data.range );
+		},
+
+		/**
+		 * The function registered by the {@link #attach} method as the
+		 * {@link CKEDITOR.plugins.textWatcher#unmatched} event listener.
+		 *
+		 * @param {CKEDITOR.eventInfo} evt
+		 */
+		onTextUnmatched: function() {
+			// Remove query and request id to avoid opening view for invalid callback (#1984).
+			this.model.query = null;
+			this.model.lastRequestId = null;
+
+			this.close();
+		}
+	};
+
+	/**
+	 * Class representing the autocomplete view.
+	 *
+	 * In order to use a different view, implement a new view class and override
+	 * the {@link CKEDITOR.plugins.autocomplete#getView} method.
+	 *
+	 * ```javascript
+	 *	myAutocomplete.prototype.getView = function() {
+	 *		return new myView( this.editor );
+	 *	};
+	 * ```
+	 *
+	 * You can also modify this class on the fly.
```

Comment:
It's not the class that is being modified here, but an instance. Slightly hacky snippet but I guess it's still acceptable.